content/blog/nodepool-upgrades/index.md

```diff
@@ -0,0 +1,161 @@
+
+### Standalone Pods
+
+A pod is the smallest deployable object in Kubernetes. It represents a single instance of an application running in your cluster. Pods are ephemeral; if a pod is evicted from a node, the pod does not replace itself. Because pods are not self-healing, it is not recommended that you create individual Pods directly. Instead, use a controller such as a Deployment to create and manage the Pod for you.
+
+To minimize downtime, ensure that all your pods are managed by a ReplicaSet, Deployment, StatefulSet or something similar. Standalone pods might need to be manually rescheduled after an upgrade.
+
+### Daemonsets
+
+A [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) ensures that all (or some) nodes run a copy of a pod. Daemonsets are often used for node monitoring or log collection and do not typically serve traffic. For these use cases, it’s usually acceptable to have a small gap in data during the worker node upgrade.
+
+### Deployments
+
+Most of the pods in your cluster are likely controlled by a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/). A Deployment represents a set of identical pods with no unique identities. A Deployment increases availability by managing multiple replicas of your application and deploying replacements if any instance fails.
+
+To eliminate downtime, ensure that your applications have a [PodDisruptionBudget](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) (PDB). A PDB helps provide higher availability by limiting the number of pods of a replicated application that are down simultaneously.
+
+For example, the following PDB declares that 80% of the pods with the `front-end` label must be available during a disruption (such as our upgrade). This ensures that the number of replicas serving load never falls below a certain percentage of the total replicas.
+
+```yaml
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: demo
+spec:
+  minAvailable: 80%
+  selector:
+    matchLabels:
+      name: front-end
+```
+
+Note that you’ll need to ensure that there are more than one replica (at least temporarily, during the upgrade) in order for the nodes to be able to be upgraded.
+
+### StatefulSets
+
+A [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset) is the Kubernetes controller type used to manage stateful applications, such as databases or messaging queues. Upgrading StatefulSets requires more consideration than upgrading Deployments.
+
+To eliminate downtime, ensure that you have configured the following:
+
+1. Add a PodDisruptionBudget (see the explanation in the previous section). For quorum-based applications, ensure that the number of replicas running is never brought below the number needed for quorum (e.g, `minAvailable: 51%`).
+
```

Comment:
spacing here looks to wide in preview. also I might put Daemonset after this one as well

Reply:
Can you clarify what you mean? I don't understand.